### DISCUSSION TITLE
自己从头开始训练SiamRPN++,但是出来的结果只能tracking小一点的目标

### DISCUSSION THREAD

**Discussion by StrugglingForBetter** [STVIR/pysot#421] [General]
自己从头开始训练SiamRPN++,但是出来的结果只能tracking小一点的目标，比如目标是一个行人，貌似就会tracking出错，请问是什么原因？由于设备问题，我只训练了VID 和 coco数据？真的是因为数据问题吗？会不会有其他问题？

**Comment by ZhiyuanChen** [comment]
出什么错误呢？

**Comment by StrugglingForBetter** [comment]
程序本身并不会出错，就是假如框住整个人，可能过段时间，框就不对了，框会变形，和原来shape不匹配，然后框的位置也会错，就是会跟丢。但是如果目标是比较小的比如人脸，或者比较小的物体，就track得挺不错，很少会跟丢

**Comment by ZhiyuanChen** [comment]
> 程序本身并不会出错，就是假如框住整个人，可能过段时间，框就不对了，框会变形，和原来shape不匹配，然后框的位置也会错，就是会跟丢。但是如果目标是比较小的比如人脸，或者比较小的物体，就track得挺不错，很少会跟丢

这个看上去不是一个代码或者网络层面的问题，很抱歉我无法帮到你。
但是我可以告诉你如何去解决问题：
首先定位问题位置，学习在框在什么样的情况下会发生变形
然后大胆提出假设，假设框的变形是由某种原因导致的
最后仔细验证假设，通过尝试不同的方法来排除假设或者解决假设
希望这是有帮助的

**Comment by StrugglingForBetter** [comment]
框会在目标范围偏大的情况下变形，然后位置也会跟丢，但是应该是模型本身的问题，inference程序本身应该没有问题，因为如果是小目标就track得比较好

**Comment by ZhiyuanChen** [comment]
> 框会在目标范围偏大的情况下变形，然后位置也会跟丢，但是应该是模型本身的问题，inference程序本身应该没有问题，因为如果是小目标就track得比较好

你的描述过于抽象，我无法从中提取到任何有价值的信息。第几帧开始跟丢？不同种类的对象的跟丢分布有什么特征？不同尺寸对象的跟丢分布又有什么特征？模型输出的置信度是如何分布的？跟丢帧和模板对比有哪些区别？重排列对排名造成了什么样的影响？

**Comment by StrugglingForBetter** [comment]
如果你可以给我一个通讯方式，我可以发一个视频给你，这样就更清楚了。
